server/backends/disk_cache/disk_cache.go

```diff
@@ -35,23 +39,60 @@ const (
 	// check the cache size.
 	janitorCheckPeriod = 100 * time.Millisecond
 
-	defaultPartition = "default"
+	defaultPartitionID = "default"
 )
 
 // DiskCache stores data on disk as files.
 // It is broken up into partitions which are independent and maintain their own LRUs.
 type DiskCache struct {
+	env               environment.Env
+	partitions        map[string]*partition
+	partitionMappings []config.DiskCachePartitionMapping
+	// The currently selected partition. Initialized to the default partition.
+	// WithRemoteInstanceName can create a new cache accessor with a different selected partition.
 	partition *partition
 	prefix    string
 }
 
-func NewDiskCache(rootDir string, maxSizeBytes int64) (*DiskCache, error) {
-	p, err := newPartition(defaultPartition, rootDir, maxSizeBytes)
-	if err != nil {
-		return nil, err
+func NewDiskCache(env environment.Env, config *config.DiskConfig, defaultMaxSizeBytes int64) (*DiskCache, error) {
+	partitions := make(map[string]*partition)
+	var defaultPartition *partition
+	for _, pc := range config.Partitions {
+		maxSizeBytes, err := units.FromHumanSize(pc.MaxSize)
```

Comment:
This is a big improvement -- but it strikes me as a little error prone to have just this one field specified in this way. WDYT about sending a separate CL that just does this for all size values specified via config?

Reply:
Sounds good.